sympy/matrices/matrices.py

```diff
@@ -3433,6 +3433,99 @@ def _eval_matrix_exp_jblock(self):
         from .sparsetools import banded
         return self.__class__(banded(size, bands))
 
+
+
+    def analytic_func(self, f, x, parameter=None):
+        """
+        Computes f(A) where A is a Square Matrix
+        and f is an analytic function.
+
+        Examples
+        ========
+
+        >>> from sympy import Symbol, Matrix, exp, S, log
+
+        >>> x = Symbol('x')
+        >>> m = Matrix([[S(5)/4, S(3)/4], [S(3)/4, S(5)/4]])
+        >>> f = log(x)
+        >>> m.analytic_func(f, x)
+        Matrix([
+        [     0, log(2)],
+        [log(2),      0]])
+
+        Parameters
+        ==========
+
+        f : Analytic Function
+        x : parameter of f
+        parameter : If Symbol('t') then it returns f(A*t) where t is parameter
```

Comment:
You should make denote the parameters with `name : type` notation and put any explanation below indented block, or it will look ugly in the HTML.

```
        f : Expr
            An analytic expression.
```